feat: adds new proxy transform attributes (#243)
Basis-Theory/docs: source/includes/api-reference/_proxies.md

| `require_auth`       | *boolean*                                            | Require a `BT-API-KEY` request header for authentication and authorization                                                                                                                                                                   |
| `request_transform`  | *[Proxy Transform](#proxies-proxy-transform-object)* | (Optional) Transform code which will be executed against the proxy's request before sending to the destination.                                                                                                                              |
| `response_transform` | *[Proxy Transform](#proxies-proxy-transform-object)* | (Optional) Transform code which will be executed against the destination's response before returning the proxy response.                                                                                                                     |

> | `application_id`     | *string*                                             | (Optional) This Application's permissions are injected into a pre-configured <a href="https://www.npmjs.com/package/@basis-theory/basis-theory-js" target="_blank">Node.js BasisTheory</a> instance passed into the Proxy Transform runtime. |

Reviewer: Should we phrase this without mentioning permissions since applications can be defined with permissions OR rules? Maybe saying API key is clearer?

```suggestion
| `application_id`     | *string*                                             | (Optional) This Application's API key is injected into a pre-configured <a href="https://www.npmjs.com/package/@basis-theory/basis-theory-js" target="_blank">Node.js BasisTheory</a> instance passed into the Proxy Transform runtime. |
```

| `configuration` | *object* | The configuration defined for the [Proxy object](#proxies-proxy-object)                                                                                                     |

### Transform Response Object

| Attribute | Type     | Description                                      |
| --------- | -------- | ------------------------------------------------ |
| `body`    | *object* | Processed body for the proxy request/response    |
| `headers` | *object* | Processed headers for the proxy request/response |

Transform Code is written in Javascript (targeting Node.js v16) and generally follows the structure:

<div class="center-column"></div>
```js
module.exports = async function (req) {
  const { body, headers, bt } = req.args; // access any args provided with the transform request
  const { MY_CONFIG } = req.configuration; // access any static config defined on the Proxy

  // do anything here!
  const processedBody = ...; // do something with body
  const processedHeaderes = ...; // do something with headers

  return {
      body: processedBody, // Proxy request/response body

>       headers: processedHeaderes // Proxy request/response headers

Reviewer: ```suggestion
      headers: processedHeaders // Proxy request/response headers
```
